File: SchoolData/src/main/java/org/gb/homework/schooldata/controller/AppController.java
package org.gb.homework.schooldata.controller;


import org.gb.homework.schooldata.AppConst;
import org.gb.homework.schooldata.service.UserService;
import org.gb.homework.schooldata.view.AssignView;
import org.gb.homework.schooldata.view.SchoolDataView;
import org.gb.homework.schooldata.view.StudentView;
import org.gb.homework.schooldata.view.TeacherView;

import java.util.Map;

public class AppController {
    private final Map<String, UserService> services;

    public AppController(Map<String, UserService> services) {
        this.services = services;
    }

    public void run() {

        var view = new SchoolDataView();

        var isWorking = true;

        while (isWorking) {

            var numberTask = view.showMenu();

            switch (numberTask) {
                case 0 -> isWorking = false;
                case 1 -> {
                    var teacherController = new TeacherController(services.get(AppConst.TEACHER), new TeacherView());
                    teacherController.run();
                }
                case 2 -> {
                    var studentController = new StudentController(services.get(AppConst.STUDENT), new StudentView());
                    studentController.run();
                }
                case 3 -> {
                    var assignController = new AssignController(services.get(AppConst.TEACHER), services.get(AppConst.STUDENT), new AssignView());
                    assignController.run();
                }
                default -> view.showTaskError();
            }
        }

        view.showWorkComplete();
    }
}

File: csharp/SchoolData/Controllers/AppController.cs
using Gb.Homework.SchoolData.Model;
using Gb.Homework.SchoolData.Service;
using Gb.Homework.SchoolData.View;

namespace Gb.Homework.SchoolData.Controllers
{


    public class AppController
    {
        private readonly Dictionary<string, IUserService> services;
        private readonly SchoolDataView view;

        public AppController(Dictionary<string, IUserService> services, SchoolDataView view)
        {
            this.services = services;
            this.view = view;
        }

        public void Run()
        {
            var isWorking = true;

            while (isWorking)
            {

                var numberTask = view.ShowMenu();

                switch (numberTask)
                {
                    case 0:
                        isWorking = false;
                        break;
                    case 1:
                        var teacherController = new TeacherController((ISchoolUserService<Teacher>)services[AppConst.TEACHER], new TeacherView());
                        teacherController.Run();
                        break;
                    case 2:
                        var studentController = new StudentController((ISchoolUserService<Student>)services[AppConst.STUDENT], new StudentView());
                        studentController.Run();
                        break;
                    case 3:
                        var teacherService = (ISchoolUserService<Teacher>)services[AppConst.TEACHER];
                        var studentService = (ISchoolUserService<Student>)services[AppConst.STUDENT];
                        var assignController = new AssignController(teacherService, studentService, new AssignView());
                        assignController.Run();
                        break;
                    default:
                        view.showTaskError();
                        break;
                }
            }

            view.showWorkComplete();
        }
    }

}
